planning/behavior_path_planner/src/utilities.cpp

```diff
@@ -1679,6 +1679,103 @@ PathWithLaneId getCenterLinePath(
+  double s_forward = s + forward_path_length;
+
+  const double buffer = parameter.backward_length_buffer_for_end_of_lane +
+                        additional_length;  // buffer for min_lane_change_length
+  const int num_lane_change =
+    std::abs(route_handler.getNumLaneToPreferredLane(lanelet_sequence.back()));
+  const double lane_length = lanelet::utils::getLaneletLength2d(lanelet_sequence);
+  const double lane_change_buffer =
+    num_lane_change * (parameter.minimum_lane_change_length + buffer);
+
+  if (route_handler.isDeadEndLanelet(lanelet_sequence.back())) {
+    s_forward = std::min(s_forward, lane_length - lane_change_buffer);
+  }
+
+  if (route_handler.isInGoalRouteSection(lanelet_sequence.back())) {
+    const auto goal_arc_coordinates =
+      lanelet::utils::getArcCoordinates(lanelet_sequence, route_handler.getGoalPose());
+    s_forward = std::min(s_forward, goal_arc_coordinates.length - lane_change_buffer);
+  }
+
+  return route_handler.getCenterLinePath(lanelet_sequence, s_backward, s_forward, true);
+}
+
+bool checkLaneIsInIntersection(
+  const RouteHandler & route_handler, const PathWithLaneId & ref,
+  const lanelet::ConstLanelets & lanelet_sequence, const double & lane_change_buffer,
+  double & additional_length_to_add)
+{
+  if (lanelet_sequence.empty()) {
+    return false;
+  }
+  const auto & path_points = ref.points;
+  const auto last_arc_coordine =
+    lanelet::utils::getArcCoordinates(lanelet_sequence, path_points.back().point.pose);
+  const auto last_arc_length = last_arc_coordine.length;
+  auto checking = path_points.back().point.pose;
+  for (auto itr = path_points.crbegin(); itr != path_points.crend(); ++itr) {
+    const auto arc_coordinate =
+      lanelet::utils::getArcCoordinates(lanelet_sequence, itr->point.pose);
+    const auto diff = last_arc_length - arc_coordinate.length;
+    if (diff >= lane_change_buffer) {
+      checking = itr->point.pose;
+      break;
+    }
+  }
+
+  lanelet::ConstLanelet check_lane;
+  if (!route_handler.getClosestLaneletWithinRoute(checking, &check_lane)) {
+    return false;
+  }
+
+  lanelet::ConstLanelets prev_lanelets;
+  if (!route_handler.getPreviousLaneletsWithinRoute(check_lane, &prev_lanelets)) {
+    return false;
+  }
+
+  const auto isHaveNeighborWithTurnDirection = [&](const lanelet::ConstLanelets & lanes) noexcept {
+    return std::any_of(lanes.cbegin(), lanes.cend(), [](const lanelet::ConstLanelet & lane) {
+      return lane.hasAttribute("turn_direction");
+    });
+  };
+
+  const auto lanes = route_handler.getNextLanelets(prev_lanelets.back());
+  if (isHaveNeighborWithTurnDirection(lanes)) {
+    lanelet::ConstLanelets lane_of_interest{check_lane};
+    // additional_length_to_add = lanelet::utils::getLaneletLength2d(check_lane);
+    for (auto itr = prev_lanelets.crbegin(); itr != prev_lanelets.crend(); ++itr) {
+      if (!itr->hasAttribute(lanelet::AttributeNamesString::LaneChange)) {
```

Comment:
If the value of this attribute is `"yes"`, it can change the lane.
So, please check the value here.

Reply:
I have change the latest commit to check the left line string and right linestring to check the has attribute value `yes`